modules/gdscript/gdscript_parser.cpp

```diff
@@ -4157,6 +4162,39 @@ bool GDScriptParser::onready_annotation(AnnotationNode *p_annotation, Node *p_ta
 	return true;
 }
 
+bool GDScriptParser::tool_button_annotation(AnnotationNode *p_annotation, Node *p_node, ClassNode *p_class) {
+#ifdef TOOLS_ENABLED
+	AnnotationNode *annotation = const_cast<AnnotationNode *>(p_annotation);
+	FunctionNode *func = static_cast<FunctionNode *>(p_node);
```

Comment:
```suggestion
	ERR_FAIL_COND_V_MSG(p_target->type != Node::FUNCTION, false, R"("@tool_button" annotation can only be applied to functions.)");
	
	AnnotationNode *annotation = const_cast<AnnotationNode *>(p_annotation);
	FunctionNode *func = static_cast<FunctionNode *>(p_node);
```

Reply:
I'm not opposed to this change exactly, but I cannot get it to hit. There are other mechanisms preventing this, like `GDScriptParser::AnnotationNode::applies_to(p_target_kinds)`.

Can you give me some hints as to when this would break, and its necessity, to help my understanding?

Reply:
These checks are present in the implementations of other annotations, so this is necessary for consistency.